internal/literals/literals.go

```diff
@@ -77,59 +77,92 @@ func Obfuscate(file *ast.File, info *types.Info, fset *token.FileSet, linkString
 			return true
 		}
 
-		if node, ok := node.(*ast.CompositeLit); ok {
-			if len(node.Elts) == 0 || len(node.Elts) > maxSizeBytes {
+		switch node := node.(type) {
+		case *ast.UnaryExpr:
+			// Account for the possibility of an ampersand prefix when
+			// passing []byte or [...]byte as an argument to a function.
+			//
+			// See issue #520.
+
+			if node.Op != token.AND {
 				return true
 			}
 
-			byteType := types.Universe.Lookup("byte").Type()
-
-			var arrayLen int64
-			switch y := info.TypeOf(node.Type).(type) {
-			case *types.Array:
-				if y.Elem() != byteType {
-					return true
+			if child, ok := node.X.(*ast.CompositeLit); ok {
+				if newnode, ok := handleCompositeLiteral(true, child, info); ok {
+					cursor.Replace(newnode)
 				}
+			}
+		case *ast.CompositeLit:
+			parent, ok := cursor.Parent().(*ast.UnaryExpr)
+			if ok && parent.Op == token.AND {
+				return true
+			}
 
-				arrayLen = y.Len()
+			if newnode, ok := handleCompositeLiteral(false, node, info); ok {
+				cursor.Replace(newnode)
+			}
+		}
 
-			case *types.Slice:
-				if y.Elem() != byteType {
-					return true
-				}
+		return true
+	}
 
-			default:
-				return true
-			}
+	return astutil.Apply(file, pre, post).(*ast.File)
+}
 
-			data := make([]byte, 0, len(node.Elts))
+// handleCompositeLiteral checks if the input node is []byte or [...]byte and
+// calls the appropriate obfuscation method, returning a new node that should
+// be used to replace it.
+//
+// If the input is not a byte slice or array, the node is returned as-is and
+// the second return value will be false.
+func handleCompositeLiteral(ref bool, node *ast.CompositeLit, info *types.Info) (ast.Node, bool) {
```

Comment:
perhaps call the bool `asPointer` for clarity

Reply:
also, what is the boolean return providing? it seems to be an `ok bool`, so perhaps use `node != nil` rather than `ok` to avoid the boolean return parameter.